keras_core/ops/math.py

```diff
@@ -450,3 +509,193 @@ def fft2(a):
+    x = backend.math.frame(x, fft_length, frame_step)
+    if window is not None:
+        if isinstance(window, str):
+            window = scipy.signal.get_window(window, frame_length)
+        win = backend.core.convert_to_tensor(window, dtype=x.dtype)
+        if len(win.shape) != 1 or win.shape[-1] != frame_length:
+            raise ValueError(
+                "The shape of `window` must be equal to [frame_length]."
+                f"Received: window shape={win.shape}"
+            )
+        l_pad = (fft_length - frame_length) // 2
+        r_pad = fft_length - frame_length - l_pad
+        win = backend.numpy.pad(win, [[l_pad, r_pad]])
+    else:
+        win = backend.numpy.ones((frame_length,), dtype=x.dtype)
+    x = backend.numpy.multiply(x, win)
+    return backend.math.rfft(x, fft_length)
+
+
+class STFT(Operation):
+    def __init__(
+        self, frame_length, frame_step, fft_length, window="hann", center=True
+    ):
+        super().__init__()
+        self.frame_length = frame_length
+        self.frame_step = frame_step
+        self.fft_length = fft_length
+        self.window = window
+        self.center = center
+
+    def compute_output_spec(self, x):
+        if len(x.shape) < 1:
+            raise ValueError(
+                f"Input should have rank >= 1. "
+                f"Received: input shape = {x.shape}"
+            )
+        if x.shape[-1] is not None:
+            padded = 0 if self.center is False else (self.fft_length // 2) * 2
+            num_frames = (
+                1 + (x.shape[-1] + padded - self.fft_length) // self.frame_step
+            )
+        else:
+            num_frames = None
+        new_shape = x.shape[:-1] + (num_frames, self.fft_length // 2 + 1)
+        return (
+            KerasTensor(shape=new_shape, dtype=x.dtype),
+            KerasTensor(shape=new_shape, dtype=x.dtype),
+        )
+
+    def call(self, x):
+        return _stft(
+            x,
+            frame_length=self.frame_length,
+            frame_step=self.frame_step,
+            fft_length=self.fft_length,
+            window=self.window,
+            center=self.center,
+        )
+
+
+@keras_core_export("keras_core.ops.stft")
+def stft(x, frame_length, frame_step, fft_length, window="hann", center=True):
+    """Computes the short-time Fourier transform along the last axis of the
+    input.
+
+    Args:
+        x: Input tensor.
+        frame_length: An integer representing the frame length in samples.
+        frame_step: An integer representing the frame hop size in samples.
+        fft_length: An integer representing the size of the FFT to apply. If not
+            specified, uses the smallest power of 2 enclosing `frame_length`.
+        window: A string or the tensor of the window. If `window` is a string,
+            it is passed to `scipy.signal.get_window` to generate the window
+            values, which are DFT-even by default. See
```

Comment:
The docstring should be standalone and should not refer to other sources to explain what argument values are valid. They should be listed here.

Reply:
I have updated the docstring to guide users to select either `"hann"` or `"hamming"`
- tf & torch: only provides hann and hamming window
- numpy & jax: https://docs.scipy.org/doc/scipy/reference/generated/scipy.signal.get_window.html